bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -821,34 +841,37 @@ protected void release(boolean wait, boolean writer)
       final int refCount = this.refCount.decrementAndGet();
       if (canEvict(refCount, writer)) {
         assert (next != null);
-        final Runnable storer = getStorer(data, readingOffset, writingOffset, storage);
-        if (future != null && future.cancel(false)) {
-          logger.debug("Block {} future is cancelled", this);
-        }
-        final int numberOfInMemBlockPermits = DataList.this.numberOfInMemBlockPermits.get();
-        if (wait && numberOfInMemBlockPermits == 0) {
-          future = null;
-          storer.run();
-        } else if (numberOfInMemBlockPermits < MAX_COUNT_OF_INMEM_BLOCKS / 2) {
-          future = storageExecutor.submit(storer);
+        if (storage == null) {
+          final Runnable storer = getStorer(data, readingOffset, writingOffset, storage);
+          if (future != null && future.cancel(false)) {
+            logger.debug("Block {} future is cancelled", this);
+          }
+          final int numberOfInMemBlockPermits = DataList.this.numberOfInMemBlockPermits.get();
+          if (wait && numberOfInMemBlockPermits == 0) {
+            future = null;
+            storer.run();
+          } else if (numberOfInMemBlockPermits < MAX_COUNT_OF_INMEM_BLOCKS / 2) {
+            future = storageExecutor.submit(storer);
+          } else {
+            future = null;
+          }
         } else {
-          future = null;
+          // Resume checks number of in memory permits, always sending max when spooling is disabled
+          resumeSuspendedClients(MAX_COUNT_OF_INMEM_BLOCKS);
```

Comment:
It is possible that the `release()` is called not for the slowest subscriber, so Publisher may still be far ahead of the slowest subscriber. Should not Publisher be resumed only if there are no subscribers behind this one?